### PR TITLE
feat(errors): improve readability of error messages

### PR DESCRIPTION
**Pull request by rbeuque74**

### CODE REVIEW

File: ovh/error.go [hunk 1]
```diff
@@ -16,10 +19,23 @@ type APIError struct {
 	QueryID string
 }
 
-func (err *APIError) Error() string {
-	if err.Class == "" {
-		return fmt.Sprintf("HTTP Error %d: %q", err.Code, err.Message)
+// Let's make sure that APIError always satisfy the Stringer and error interfaces
```
Comment (deathiop) [inline ovh/error.go:22]:
```suggestion
// Let's make sure that APIError always satisfies the fmt.Stringer and error interfaces
```

File: ovh/error.go [hunk 2]
```diff
@@ -16,10 +19,23 @@ type APIError struct {
...
+var _ fmt.Stringer = APIError{}
+var _ error = APIError{}
+
+func (err APIError) Error() string {
```
Comment (deathiop) [inline ovh/error.go:26]: This should use a `strings.Builder`

```go
func (err APIError) Error() string {
    var sb strings.Builder

    // Base message
    fmt.Fprint(&sb, "OVHcloud API error (status code ", err.Code, "): ")

    // Append class if any
    if err.Class != "" {
        fmt.Fprint(&sb, err.Class, ": ")
    }

    // Real error message, quoted
    fmt.Fprintf(&sb, "%q", err.Message)

    // QueryID if any
    if err.QueryID != "" {
        fmt.Fprint(&sb, " (X-OVH-Query-Id: ", err.QueryID, ")")
    }

    return sb.String()
}
```

File: ovh/error_test.go [hunk 3]
```diff
@@ -8,17 +8,27 @@ import (
...
-	td.CmpString(t, err, `HTTP Error 409: CartAlreadyExists: "the cart id \"foobar\" already exists" (X-OVH-Query-Id: EU.ext-99.foobar)`)
+	expected = `OVHcloud API error (status code 409): CartAlreadyExists: "the cart id \"foobar\" already exists" (X-OVH-Query-Id: EU.ext-99.foobar)`
+	td.Cmp(t, err.Error(), expected)
+	td.Cmp(t, err.String(), expected)
+
+	err.Class = ""
+	expected = `OVHcloud API error (status code 409): "the cart id \"foobar\" already exists" (X-OVH-Query-Id: EU.ext-99.foobar)`
+	td.Cmp(t, err.Error(), expected)
+	td.Cmp(t, err.String(), expected)
+
```
Comment (amstuta) [inline ovh/error_test.go:33]:
```suggestion
```

Comment (rbeuque74) [reply ovh/error_test.go:33]: My editor run linter that add a final line to every file, I guess this one is non-blocking and I will not apply.

**Review by Oso-hub — COMMENTED** [review]
Very good

**Pull request merged**